lib/Backend/FlowGraph.cpp

```diff
@@ -3695,6 +3695,24 @@ Loop::IsSymAssignedToInSelfOrParents(StackSym * const sym) const
     return false;
 }
 
+BasicBlock *
+Loop::GetTailBlock() const
+{
+    BasicBlock * tail = nullptr;
+
+    BasicBlock * loopHeader = this->GetHeadBlock();
+    FOREACH_PREDECESSOR_BLOCK(pred, loopHeader)
+    {
+        if (this->IsDescendentOrSelf(pred->loop))
+        {
+            tail = pred;
+        }
+    } NEXT_PREDECESSOR_BLOCK;
+    
+    Assert(tail);
+    return tail;
```

Comment:
What happens when there are multiple back-edges?

Reply:
The comment at the call site of this method explains why is it ok to get any tail block - basically we'll be getting the value of a PRE candidate sym on the block returned by this method. Since, the sym is a PRE candidate sym, its value must be the same on all tail blocks. I'll update the method name to GetAnyTailBlock